Add "federation" and "cluster only" prometheus configurations

stephen-soltesz opened the pull request:
To support multiple k8s clusters, this change adds federation support to the prometheus configs.

At deployment time, it is necessary to choose whether you are deploying the prometheus "cluster" configuration or the prometheus "federation" configuration, where federation is a superset of the cluster configuration.

M-Lab will need only one federation configuration per GCP project. And, every k8s cluster will deploy the prometheus cluster configuration to monitor that cluster and the services running within it.

This change also re-arranges the k8s and container configuration directories to more clearly match their use-cases.

There is some unfortunate redundancy in the configurations at this point. That is, the cluster and federation prometheus.yml configurations are largely identical, differing only in their resource requests and limits and whether they use recording rules.

My thinking at this time is that the federation instance will calculate all rules to avoid per-cluster configuration variations which may become hard to track. If calculating all recording rules at the federation instance is a problem in practice, then we can revisit this choice.

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/m-lab/prometheus-support/12)
<!-- Reviewable:end -->

pboothe commented:
LGTM

On Mon, Apr 3, 2017 at 4:36 PM, Stephen Soltesz <notifications@github.com>
wrote:

> @stephen-soltesz <https://github.com/stephen-soltesz> requested your
> review on: m-lab/prometheus-support#12
> <https://github.com/m-lab/prometheus-support/pull/12> Add "federation"
> and "cluster only" prometheus configurations.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/m-lab/prometheus-support/pull/12#event-1027222136>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAIxtktaOTaJRqTztf84_I56wA_6wuZGks5rsVgxgaJpZM4Mx_0y>
> .
>



-- 
ᴹ̶LAB <http://measurementlab.net/> | Measure the Internet, save the data,
and make it universally accessible and useful.

stephen-soltesz commented:
Thank you!

I also added a comment for the QoS calculation and added explicit resource requests and limits on the node-exporter that is part of the prometheus-server pod.
